crates/gen-host-js/src/lib.rs

```diff
@@ -421,6 +429,30 @@ impl Js {
                 }
             "),
 
+            Intrinsic::HasOwnProperty => self.src.js("
+                const hasOwnProperty = Object.prototype.hasOwnProperty;
+            "),
+
+            Intrinsic::GetErrorPayload => {
+                let hop = self.intrinsic(Intrinsic::HasOwnProperty);
+                uwrite!(self.src.js, "
+                    function getErrorPayload(e) {{
+                        if ({hop}.call(e, 'payload')) return e.payload;
+                        if ({hop}.call(e, 'message')) return String(e.message);
+                        return String(e);
+                    }}
+                ")
+            },
+
+            Intrinsic::ComponentError => self.src.js("
+                class ComponentError extends Error {
+                    constructor (payload) {
+                        super(payload);
+                        this.payload = payload;
+                    }
+                }
```

Comment:
Would it be best to remove the typescript mode for now? With the lack of import checking and new features like this aren't showing up in the typescript bindings I'm worried that the typescript support is falling pretty far behind. I was originally under the impression that typescript support was pretty important to have, but it seems like you're not using it much so it may be less important than I originally thought.

Reply:
I would say it's still very important to get TypeScript and documentation comments in TypeScript for the generated exports interface of the component, so that consumers get type hinting when importing the component. But that typing the component internals and imports could well be dropped.